waspc/src/Wasp/Generator/ServerGenerator.hs

```diff
@@ -83,15 +88,22 @@ genServer spec =
     genFileCopy = return . C.mkTmplFd
 
 genDotEnv :: AppSpec -> Generator [FileDraft]
-genDotEnv spec = return $
-  case AS.dotEnvServerFile spec of
-    Just srcFilePath
-      | not $ AS.isBuild spec ->
-          [ createCopyFileDraft
-              (C.serverRootDirInProjectRootDir </> dotEnvInServerRootDir)
-              srcFilePath
-          ]
-    _ -> []
+-- Don't generate .env if we are building for production, since .env is to be used only for
+-- development.
+genDotEnv spec | AS.isBuild spec = return []
```

Comment:
Are we sure about this? The user might still expect to get this `.env.server` copied over?

Reply:
See answer above, but short answer is that they will not expect because dotenv philosophy is what we do here + we also mention it in docs.